src/wallet/rpcwallet.cpp

```diff
@@ -4113,6 +4113,33 @@ static UniValue upgradewallet(const JSONRPCRequest& request)
     return error.original;
 }
 
+static UniValue migratewallet(const JSONRPCRequest& request)
```

Comment:
nit: might be good to use the non-deprecated constructor to avoid having to change this again. See e.g. #19528

Reply:
Getting a build error when I try to do this:

```wallet/rpcwallet.cpp:4280:5: error: no matching constructor for initialization of 'const CRPCCommand'
    { "wallet",             "migratewallet",                    &migratewallet,                 {} },
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rpc/server.h:120:5: note: candidate constructor not viable: no known conversion from 'RPCHelpMan (*)(const JSONRPCRequest &)' to 'rpcfn_type' (aka 'UniValue (*)(const JSONRPCRequest &)') for 3rd argument
    CRPCCommand(const char* category, const char* name, rpcfn_type fn, std::initializer_list<const char*> args)
    ^
./rpc/server.h:107:5: note: candidate constructor not viable: no known conversion from 'RPCHelpMan (*)(const JSONRPCRequest &)' to 'RpcMethodFnType' (aka 'RPCHelpMan (*)()') for 3rd argument
    CRPCCommand(std::string category, std::string name_in, RpcMethodFnType fn, std::vector<std::string> args_in)
    ^
./rpc/server.h:100:5: note: candidate constructor not viable: requires 5 arguments, but 4 were provided
    CRPCCommand(std::string category, std::string name, Actor actor, std::vector<std::string> args, intptr_t unique_id)
    ^
./rpc/server.h:91:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
class CRPCCommand
      ^
./rpc/server.h:91:7: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 4 were provided
1 error generated.
```

Reply:
Not sure why. This diff compiles for me:

```diff
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index cc89c4992b..1361955a4b 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -4171,9 +4171,9 @@ static UniValue upgradewallet(const JSONRPCRequest& request)
     return error.original;
 }
 
-static UniValue migratewallet(const JSONRPCRequest& request)
+static RPCHelpMan migratewallet()
 {
-    RPCHelpMan{"migratewallet",
+    return RPCHelpMan{"migratewallet",
         "\nMigrate the wallet to a descriptor wallet.\n"
         "A new wallet backup will need to be made.",
         {},
@@ -4181,9 +4181,9 @@ static UniValue migratewallet(const JSONRPCRequest& request)
         RPCExamples{
             HelpExampleCli("migratewallet", "")
             + HelpExampleRpc("migratewallet", "")
-        }
-    }.Check(request);
-
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
     std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
     if (!wallet) return NullUniValue;
     CWallet* const pwallet = wallet.get();
@@ -4196,6 +4196,8 @@ static UniValue migratewallet(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_WALLET_ERROR, error.original);
     }
     return error.original;
+},
+    };
 }
 
 RPCHelpMan abortrescan();

Reply:
Ah, I seem to have made a mistake somewhere.